diff --git a/node_modules/html-webpack-plugin/index.js b/node_modules/html-webpack-plugin/index.js
index 55a357d..7d21b20 100644
--- a/node_modules/html-webpack-plugin/index.js
+++ b/node_modules/html-webpack-plugin/index.js
@@ -29,6 +29,7 @@ class HtmlWebpackPlugin {
       cache: true,
       showErrors: true,
       chunks: 'all',
+      entryPoint: false,
       excludeChunks: [],
       chunksSortMode: 'auto',
       meta: {},
@@ -106,7 +107,13 @@ class HtmlWebpackPlugin {
         timings: false,
         version: false
       };
-      const allChunks = compilation.getStats().toJson(chunkOnlyConfig).chunks;
+      let allChunks = compilation.getStats().toJson(chunkOnlyConfig).chunks;
+      // if we have an entry point defined
+      if (self.options.entryPoint) {
+        const entryChunks = compilation.getStats().toJson(chunkOnlyConfig).entrypoints;
+        // get just chunks for the entry point
+        allChunks = allChunks.filter(c => entryChunks[self.options.entryPoint].chunks.includes(c.id));
+      }
       // Filter chunks (options.chunks and options.excludeCHunks)
       let chunks = self.filterChunks(allChunks, self.options.chunks, self.options.excludeChunks);
       // Sort chunks
@@ -372,10 +379,6 @@ class HtmlWebpackPlugin {
   filterChunks (chunks, includedChunks, excludedChunks) {
     return chunks.filter(chunk => {
       const chunkName = chunk.names[0];
-      // This chunk doesn't have a name. This script can't handled it.
-      if (chunkName === undefined) {
-        return false;
-      }
       // Skip if the chunk should be lazy loaded
       if (typeof chunk.isInitial === 'function') {
         if (!chunk.isInitial()) {
@@ -385,11 +388,11 @@ class HtmlWebpackPlugin {
         return false;
       }
       // Skip if the chunks should be filtered and the given chunk was not added explicity
-      if (Array.isArray(includedChunks) && includedChunks.indexOf(chunkName) === -1) {
+      if (Array.isArray(includedChunks) && chunkName && includedChunks.indexOf(chunkName) === -1) {
         return false;
       }
       // Skip if the chunks should be filtered and the given chunk was excluded explicity
-      if (Array.isArray(excludedChunks) && excludedChunks.indexOf(chunkName) !== -1) {
+      if (Array.isArray(excludedChunks) && chunkName && excludedChunks.indexOf(chunkName) !== -1) {
         return false;
       }
       // Add otherwise
